src/smc-webapp/smc_chat.cjsx

```diff
@@ -323,27 +323,86 @@ Message = rclass
                 {cols}
             </Row>
 
+sortCopy = (arr) ->
+    return arr.slice(0).sort()
+
+are_dicts_equal = (a, b) ->
+    # assuming maps to lists
+    sorteda = {}
+    for key in sortCopy(Object.keys(a))
+        sorteda[key] = a[key]
+    sortedb = {}
+    for key in sortCopy(Object.keys(b))
+        sortedb[key] = b[key]
+    return JSON.stringify(sorteda) == JSON.stringify(sortedb)
+
+diffArray = (arr1, arr2) ->
```

Comment:
Please comment what this does. Reading the code, it doesn't seem to do any of my first 3 guesses.